src/ORM/Connect/DBSchemaManager.php

```diff
@@ -454,16 +454,22 @@ public function requireTable(
      */
     public function dontRequireTable($table)
     {
-        if (isset($this->tableList[strtolower($table)])) {
-            $suffix = '';
-            while (isset($this->tableList[strtolower("_obsolete_{$table}$suffix")])) {
-                $suffix = $suffix
-                        ? ((int)$suffix + 1)
-                        : 2;
-            }
-            $this->renameTable($table, "_obsolete_{$table}$suffix");
-            $this->alterationMessage("Table $table: renamed to _obsolete_{$table}$suffix", "obsolete");
+        if (!isset($this->tableList[strtolower($table)])) {
+            return;
+        }
+
+        if (Config::inst()->get(static::class, 'fix_table_case_on_build')) {
+            $this->fixTableCase($table);
```

Comment:
This will result in a large number of renames.

`SiteTree_versions -> somerandomtable -> SiteTree_Versions -> _obsolete_SiteTree_Versions`

Wouldn't a simpler fix be.

```php
$renameFrom = $this->tableList[strtolower($table)];
$renameTo = "_obsolete_{$table}$suffix";
$this->renameTable($renameFrom, $renameTo);
```